app/services/submission_service.rb

```diff
@@ -19,7 +19,14 @@ def create_submission(
       ]
       user =
         if gravity_user_id
-          User.find_or_create_by!(gravity_user_id: gravity_user_id)
+          convection_user =
```

Comment:
Great, but if-else does almost the same thing, can we refactor something like:
```
      user = User.find_or_create_by!(gravity_user_id: gravity_user_id) if gravity_user_id
      user ||= User.create!
      user.update(
        name: submission_params[:user_name],
        email: submission_params[:user_email],
        phone: submission_params[:user_phone],
        session_id: submission_params.delete(:session_id)
      )
      create_params = submission_params.merge(user_id: user.id)
```

Reply:
Another question is whether it is necessary to check this data for presence? I mean, when creating a submission, this data will overwrite the already existing data about the user, and even if it does not come, they will still overwrite it.

Reply:
Thanks for noticing that there is a refactoring opportunity here! I agree that (1) the if-else is not necessary, and (2) we won't need to find existing users once there is a 1:1 relationship between submission and user records.

I will close this PR so that you can do this refactoring in #1179